Static style guide

danbeere opened the pull request:
Still a WIP, but this is the initial commit of our static style guide
for inner.

KunalJain commented:
Looks great! @danbeere 

Few things to consider:
- Picker pattern than a branch picker
- List pattern than a branch list. We should be able to use that list for deploy etc in future. 
- Same as above for commit list.
- Patterns for graphs than Insights
- CTA's on graphs
- Patterns for pages - Example Add project, build list, project settings etc.

Peeja commented:
This looks fantastic. I only have one concern: I'd strongly prefer if we divorce the notion of our heading sizes from the HTML tag names `h1`–`h6`. I think it's much better not to imply that those headings need to be implemented with those HTML elements. Currently, we have a lot of style attached to those tag names, and we end up overriding it all over the place.

I think it's also semantically valuable to find an actual name for them. For instance, these:

![75d45042-0559-11e6-864b-43c50c531365 png 2880x1800](https://cloud.githubusercontent.com/assets/2407/14615644/aadfdd0e-0574-11e6-905f-7d1543f02636.png)

aren't "H2s", they're "Module Headings" (or "Primary Module Headings" or "Module Titles" or something else appropriate). Finding that name for them is valuable. Case in point:

![75d45042-0559-11e6-864b-43c50c531365 png 2880x1800 -1](https://cloud.githubusercontent.com/assets/2407/14615685/e0c6c162-0574-11e6-86a1-e1ed3c801640.png)

I'm not actually sure what this heading is. It's called an "H3", but that doesn't tell me what it means or when we use one.

Speaking of headings:

![skitched-20160418-145228](https://cloud.githubusercontent.com/assets/2407/14615754/2c8c234e-0575-11e6-991d-84b0ee910e13.png)

I'm a stickler for names, but I think it's important: these are headings, not headers. (A header is a block of content at the beginning of some larger unit of content; a heading is a line of text that serves as a title in the content outline.) I'd also prefer we called them something like:

* Heading Biggest
* Heading Big
* Heading Medium
* Heading Small
* Heading Smallest

again, so we don't conflate their names with the HTML names `h1` though `h5`, which won't necessarily be how they're always implemented. (Incidentally, I'd say it's appropriate to call one of these "Heading Big" here rather than "Module Heading", just as it's appropriate to call one of our colors `@blueDark`: this is our typography palette, just as we have a color palette. We'd say, therefore, that the Module Heading is a Heading Big, just as we'd say that External Links are `@blueDark`.)

Peeja commented:
Oh, two other things:

* `@blueDarkest` is missing a hex value.
* Modules (I'm glad to have that word, by that way), Build List Items, and each project in Insights are all instances of what we've been calling a Card; it might be useful to see the Card style demonstrated abstractly as well.

danbeere commented:
Static style guide [iteration](https://github.com/circleci/media/pull/5/commits/db54fe8dff3cac0bdb80719d3a7ac5d4d501decc) addressing some feedback of the previous commit. Some screenshots below.

<img width="1440" alt="screen shot 2016-04-18 at 3 02 30 pm" src="https://cloud.githubusercontent.com/assets/9681658/14621360/fc5a0d2e-0576-11e6-80ea-b954efd4ade8.png">

<img width="1440" alt="screen shot 2016-04-18 at 3 03 36 pm" src="https://cloud.githubusercontent.com/assets/9681658/14621364/037bbd28-0577-11e6-8b8f-c226dd256b0d.png">

<img width="1440" alt="screen shot 2016-04-18 at 3 01 39 pm" src="https://cloud.githubusercontent.com/assets/9681658/14621367/0a4429ec-0577-11e6-8106-fee7c1044173.png">

<img width="1440" alt="screen shot 2016-04-18 at 3 01 33 pm" src="https://cloud.githubusercontent.com/assets/9681658/14621372/0e9101d2-0577-11e6-90b6-2ad525dbc2bd.png">

danbeere commented:
[Update](https://github.com/circleci/media/pull/5/commits/0e4ea43164618fc4c77f12ac1a49025ed2ccf9e1) required to add some space between elements for readability. Some screenshots below. 

@KunalJain, I'll continue to add more usage examples and if you've any recommendations that'd be great too. Cheers 🍻 

<img width="1440" alt="screen shot 2016-04-18 at 5 21 43 pm" src="https://cloud.githubusercontent.com/assets/9681658/14623971/3b907258-058b-11e6-97fe-d01a401b6cd9.png">

<img width="1440" alt="screen shot 2016-04-18 at 5 24 40 pm" src="https://cloud.githubusercontent.com/assets/9681658/14623973/4058ee3c-058b-11e6-8e17-400e5e477270.png">

Peeja commented:
![037bbd28-0577-11e6-8b8f-c226dd256b0d png 2880x1800](https://cloud.githubusercontent.com/assets/2407/14644629/af607102-0620-11e6-8e3a-eca21741f108.png)

Because these text styles are more than just sizes (the smallest two are all-caps), we should find names for the entire styles. Using `font-size: @fontSizeSmall` won't get you an all-caps heading; instead, `@fontSizeSmall` is just one attribute of a text style for a particular kind of heading, which also includes `font-weight` and, in this case, `text-transform`. We'll end up using these text style names as mixins or extends in the code.

![3b907258-058b-11e6-97fe-d01a401b6cd9 png 2880x1800](https://cloud.githubusercontent.com/assets/2407/14644902/a9c650b2-0621-11e6-903d-aeb8df2406ec.png)

I'm concerned that this doesn't demonstrate a "List B" (let's find a more meaningful name for that, too) so much as a Commit List. Reading this, I wouldn't know how to reuse this element for anything else. The other place where we have a "list with items that can be expanded or collapsed" that I can think of is the Test Failures, which have only the "More" toggle in common with this design:

![circleci_circle 140567 - circleci](https://cloud.githubusercontent.com/assets/2407/14645116/765868d6-0622-11e6-9513-b19b2fae1339.png)

If that's really all that a "List B" dictates, that's not clear to me from the way it's laid out here: it reads to me as if the header and the individual rows are significant.

(Also, there still an "H5" in the List B header.)

danbeere commented:
[Updated](https://github.com/circleci/media/pull/5/commits/0cd92bf56251374052ac302b6c0cf7a326b75798) lists section to include our tests failure pattern. 
Screenshot below:

<img width="1440" alt="screen shot 2016-04-19 at 1 12 05 pm" src="https://cloud.githubusercontent.com/assets/9681658/14654067/c84ceae6-0630-11e6-992e-3cd8affc04e6.png">

Thanks a million for the feedback, @Peeja! Didn't update the font styles section as part of this, but below is an example that specifies font styles including all-caps. Is that what you have in mind or something else?

<img width="350" alt="screen shot 2016-04-19 at 1 16 23 pm" src="https://cloud.githubusercontent.com/assets/9681658/14654228/78e99bec-0631-11e6-91e8-422203bf8d78.png">

Cc @KunalJain

danbeere commented:
Style guide updates based on PM feedback to include tables, combo buttons, pagination, messaging templates and tooltips, as well as usage descriptions for some patterns. Cc @KunalJain 

<img width="1440" alt="screen shot 2016-04-19 at 6 13 14 pm" src="https://cloud.githubusercontent.com/assets/9681658/14660520/f2a0714e-065a-11e6-9041-2d9a2393b88a.png">

<img width="1440" alt="screen shot 2016-04-19 at 6 13 28 pm" src="https://cloud.githubusercontent.com/assets/9681658/14660525/f8f8f606-065a-11e6-8b74-0abb5a1205cc.png">

<img width="1440" alt="screen shot 2016-04-19 at 6 13 38 pm" src="https://cloud.githubusercontent.com/assets/9681658/14660527/fe4fc6fc-065a-11e6-9f46-28be0defd39f.png">

<img width="1440" alt="screen shot 2016-04-19 at 6 13 52 pm" src="https://cloud.githubusercontent.com/assets/9681658/14660529/03208c5c-065b-11e6-8815-0f7fd3fa6f4f.png">

<img width="1440" alt="screen shot 2016-04-19 at 6 14 17 pm" src="https://cloud.githubusercontent.com/assets/9681658/14660531/07a0a000-065b-11e6-90f1-8a769fb56fa7.png">

Peeja commented:
Yeah, that's exactly what I'm going for. (I assume "Recipient", "Address", and "Label" are the names for the styles in this example?)

Also, the List Expandable reads much clearer to me now. Based on the name and the two examples, I can infer that we're really just talking about the expandability toggle. 👍

danbeere commented:
Add expandable list item from the stages/steps of our build page ⬇️ 

Also on Zeplin 👍 https://zpl.io/ZiOq4V

<img width="1440" alt="screen shot 2016-04-22 at 10 49 19 am" src="https://cloud.githubusercontent.com/assets/9681658/14750418/966e6572-0879-11e6-9817-897c33d55c31.png">

Peeja commented:
I'm glad to see the step component on the style guide, but I don't think it counts as a List Expandable. A List Expandable shows the first three items of a list and then a disclosure to reveal more. The step component just hides or shows everything. It's not a list at all.

danbeere commented:
@Peeja Good point. Any thoughts on how best to categorize it?

Peeja commented:
I don't think we use that component for anything else, so I think it's just a Build Step.

danbeere commented:
Update selector UI element with icon top right (optional)

<img width="1440" alt="screen shot 2016-04-24 at 3 14 31 pm" src="https://cloud.githubusercontent.com/assets/9681658/14770867/83d425fc-0a2f-11e6-91bf-527a085ff4ec.png">

danbeere commented:
@KunalJain @Peeja Reordered and renamed some UI elements to agree on. Any thoughts?

Peeja commented:
Looking good! Under List Expandable's Usage, though, I'd say, "For lists which can be expanded or collapsed." It's the list that expands, not the items.

danbeere commented:
Update - https://github.com/circleci/media/pull/5/commits/dc231e8ffdda04f58be6e3540609498540599fbb - list expandable usage copy following @Peeja's feedback

KunalJain commented:
:ship: :shipit:
